tests/model_inheritance/tests.py

```diff
@@ -72,6 +72,14 @@ def test_model_with_distinct_accessors(self):
             AttributeError, getattr, post, "attached_%(class)s_set"
         )
 
+    def test_model_with_distinct_related_query_name(self):
+        self.assertQuerysetEqual(Post.objects.filter(attached_model_inheritance_comments__is_spam=True), [])
+
+        # The Post model doesn't have a related query accessor based on
+        # related_name (attached_comment_set).
+        with self.assertRaises(FieldError):
```

Comment:
It would help to use `assertRaisesMessage` here. I always find it useful to be able to match an error message up to where it's tested.